### PR TITLE
add internal/config/tls test

### PR DESCRIPTION
**Pull request by vankichi**
### Description:
I added test of `internal/config/tls`.
And also, I refactored `internal/tls` for testing

### Environment:

- Go Version: 1.14.3
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.11.5

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [x] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/tls/option.go [hunk 1]
```diff
@@ -23,47 +23,6 @@ type Option func(*credentials) error
 
 var (
 	defaultOpts = []Option{
-		WithTLSConfig(&tls.Config{
```
Comment (kevindiu) [inline internal/tls/option.go:26]: why do we move this block to tls.go?

Comment (vankichi) [reply internal/tls/option.go:26]: To avoid same pointer between `want` and `got` in test

**Pull request merged**